packages/bff/src/model/api/converters/catalogClientApiConverter.ts

```diff
@@ -0,0 +1,211 @@
+      (e) => e.id === agreement.descriptorId
+    );
+
+    return (
+      eserviceDescriptor !== undefined &&
+      eservice.descriptors
+        .filter((d) => Number(d.version) > Number(eserviceDescriptor.version))
+        .find(
+          (d) =>
+            (d.state === descriptorApiState.PUBLISHED ||
+              d.state === descriptorApiState.SUSPENDED) &&
+            (agreement.state === agreementApiState.ACTIVE ||
+              agreement.state === agreementApiState.SUSPENDED)
+        ) !== undefined
+    );
+  };
+
+  const partialEnhancedEservice = {
+    id: eservice.id,
+    name: eservice.name,
+    description: eservice.description,
+    producer: {
+      id: eservice.producerId,
+      name: producerTenant.name,
+    },
+    isMine: isRequesterEqProducer,
+    hasCertifiedAttributes,
+  };
+
+  return {
+    ...partialEnhancedEservice,
+    ...(activeDescriptor
+      ? {
+          activeDescriptor: {
+            id: activeDescriptor.id,
+            version: activeDescriptor.version,
+            audience: activeDescriptor.audience,
+            state: activeDescriptor.state,
+          },
+        }
+      : {}),
+    ...(agreement
+      ? {
+          agreement: {
+            id: agreement.id,
+            state: agreement.state,
+            canBeUpgraded: isUpgradable(agreement),
+          },
+        }
+      : {}),
+  };
+}
+
+export function toBffCatalogApiDescriptorAttribute(
+  attributes: attributeRegistryApi.Attribute[],
+  descriptorAttributes: catalogApi.Attribute[]
+): bffApi.DescriptorAttribute[] {
+  return descriptorAttributes.map((attribute) => {
+    const foundAttribute = attributes.find((att) => att.id === attribute.id);
+    if (!foundAttribute) {
+      throw attributeNotExists(unsafeBrandId(attribute.id));
+    }
+
+    return {
+      id: attribute.id,
+      name: foundAttribute.name,
+      description: foundAttribute.description,
+      explicitAttributeVerification: attribute.explicitAttributeVerification,
+    };
+  });
+}
+
+export function toBffCatalogApiDescriptorInterface(
+  document: catalogApi.EServiceDoc
+): bffApi.EServiceDoc {
+  return {
+    id: document.id,
+    name: document.name,
+    contentType: document.contentType,
+    prettyName: document.prettyName,
+  };
+}
+
+export function toBffCatalogApiEserviceRiskAnalysis(
+  riskAnalysis: catalogApi.EServiceRiskAnalysis
+): bffApi.EServiceRiskAnalysis {
+  const answers: { [key: string]: string[] } = {};
+
+  riskAnalysis.riskAnalysisForm.singleAnswers
+    .concat(riskAnalysis.riskAnalysisForm.multiAnswers)
+    .map((answer) => ({
+      questionId: answer.questionId,
+      answer: answer.value,
+    }))
+    .forEach((QA) => {
+      if (answers[`${QA.questionId}`] && QA.answer) {
+        answers[`${QA.questionId}`] = [
+          ...answers[`${QA.questionId}`],
+          QA.answer,
+        ];
+      } else {
+        answers[`${QA.questionId}`] = [];
+      }
+    });
```

Comment:
```suggestion
    .map((answer) => ({
      questionId: answer.id,
      answer: answer.value,
    }))
    .forEach((QA) => {
      if (answers[QA.questionId] && QA.answer) {
        answers[QA.questionId] = [...answers[QA.questionId], QA.answer];
      } else {
        answers[QA.questionId] = [];
      }
    });
```
Here `answer.questionId` doesn't exist in that type, so it was inferred as `unknown`, assigning `answer.id` infers the correct type and we can also use it as index of the arrays.

P.S.: I expected an error since `answer.questionId` didn't exist on that type, maybe it was not raised because the type is defined with passthrough? cc @Carminepo2

Reply:
@AsterITA, good catch 🙌🏻 !!!
In this [commit](https://github.com/pagopa/interop-be-monorepo/pull/712/commits/392cb8cb0f657c34cd8f1fd549a8bafef3357c5c), I have replaced `forEach` with `reduce` 😋 and change the map's key used.
However, I have a doubt about which key should be added to the resulting map: `answer.id` or `answer.key`? Considering the [BFF Scala codebase](https://github.com/pagopa/interop-be-backend-for-frontend/blob/327f063053c77373eb574e511f1daf4f7b64b4b6/src/main/scala/it/pagopa/interop/backendforfrontend/service/types/CatalogProcessServiceTypes.scala#L50), the map is created using `answer.key`. Can you confirm if this is correct?

PS: Zodios `passthrough` option could be the cause of type check failure; I don't have invested lots time but using option ```strictObjects: true```  during api-clients generation in [generate.ts](https://github.com/pagopa/interop-be-monorepo/blob/392cb8cb0f657c34cd8f1fd549a8bafef3357c5c/packages/api-clients/generate.ts#L76) is not effectfull

Reply:
Yes, the key to use in the map is indeed `answer.key`

The `passthrough` problem maybe will fix itself when we will switch to a code-first implementation